internal/pkg/aws/cloudwatchlogs/cloudwatchlogs_test.go

```diff
@@ -237,6 +238,146 @@ func TestLogEvents(t *testing.T) {
 			wantLogEvents: nil,
 			wantErr:       fmt.Errorf("get log events of %s/%s: %w", "mockLogGroup", "mockLogStream", mockError),
 		},
+		"should filter out wrong prefixes": {
+			logGroupName: "mockLogGroup",
+			logStream:    []string{"copilot/"},
+			mockcloudwatchlogsClient: func(m *mocks.Mockapi) {
+				m.EXPECT().DescribeLogStreams(&cloudwatchlogs.DescribeLogStreamsInput{
+					LogGroupName: aws.String("mockLogGroup"),
+					Descending:   aws.Bool(true),
+					OrderBy:      aws.String("LastEventTime"),
+				}).Return(&cloudwatchlogs.DescribeLogStreamsOutput{
+					LogStreams: []*cloudwatchlogs.LogStream{
+						{
+							LogStreamName: aws.String("copilot/mockLogGroup/mockLogStream"),
+						},
+						{
+							LogStreamName: aws.String("states/copilot-mockLogGroup/abcde"),
+						},
+					},
+				}, nil)
+				m.EXPECT().GetLogEvents(&cloudwatchlogs.GetLogEventsInput{
+					LogGroupName:  aws.String("mockLogGroup"),
+					LogStreamName: aws.String("copilot/mockLogGroup/mockLogStream"),
+				}).Return(&cloudwatchlogs.GetLogEventsOutput{
+					Events: []*cloudwatchlogs.OutputLogEvent{
+						{
+							Message:   aws.String("some log"),
+							Timestamp: aws.Int64(0),
+						},
+					},
+				}, nil)
+			},
+			wantLogEvents: []*Event{
+				{
+					LogStreamName: "copilot/mockLogGroup/mockLogStream",
+					Message:       "some log",
+					Timestamp:     0,
+				},
+			},
+			wantLastEventTime: map[string]int64{
+				"copilot/mockLogGroup/mockLogStream": 0,
+			},
+			wantErr: nil,
+		},
+		"should limit log streams fetched": {
+			logGroupName:   "mockLogGroup",
+			logStreamLimit: aws.Int64(1),
+			mockcloudwatchlogsClient: func(m *mocks.Mockapi) {
+				m.EXPECT().DescribeLogStreams(&cloudwatchlogs.DescribeLogStreamsInput{
+					LogGroupName: aws.String("mockLogGroup"),
+					Descending:   aws.Bool(true),
+					OrderBy:      aws.String("LastEventTime"),
+					Limit:        aws.Int64(1),
+				}).Return(&cloudwatchlogs.DescribeLogStreamsOutput{
+					LogStreams: []*cloudwatchlogs.LogStream{
```

Comment:
I may be reading this wrong, but as in the next test, should this return more than one log stream so that we can confirm that only one was fetched?